Registry backed by OCS and registry library functions

ramkiperiy opened the pull request:
Updated deployment.py to change registry backend to OCS CEPHFS RWX PVC
registry.py -- registry library functions python file
ocp.py -- added new function to perform path operation in crd
constants.py -- added new constants regarding registry deployment
Requirements KNIP: https://jira.coreos.com/browse/KNIP-1080

Signed-off-by: ramkiperiy <rperiyas@redhat.com>

ramkiperiy commented on ocs_ci/ocs/ocp.py:
```diff
@@ -241,6 +241,24 @@ def apply(self, yaml_file):
         command = f"apply -f {yaml_file}"
         return self.exec_oc_cmd(command)
 
+    def patch(self, resource_name, params, type='json'):
```
This function is already part of pr #803 but this code has some enhancement, will resolve the conflict if pr #803 is merged.

ramkiperiy commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+
+def image_pull(image_url):
+    """
+    Function to pull images from repositories
```
Args:
        image_url (str): Image url container image repo link
Will add above args part, in updates

clacroix12 commented:
@ramkiperiy Is there an issue/ticket describing the requirements and reason for these changes? If so can you link it here please?

ramkiperiy commented:
> @ramkiperiy Is there an issue/ticket describing the requirements and reason for these changes? If so can you link it here please?

@clacroix12 here is the KNIP https://jira.coreos.com/browse/KNIP-1080

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+
+def get_registry_pvc():
+    """
+    Function to get registry pvc
```
Missing `Returns:` section

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+def get_default_route_name():
+    """
+    Function to get default route name
+    Return:
```
Returns: and one empty line above it

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+    Function to get default route name
+    Return:
+        route_name (str): Returns default route name
+    """
```
Here should be one empty line above

petr-balogh replied:
This is not resolved and here should be one empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+def add_role_to_user(role_type, user):
+    """
+    Function to add role to user
+    Args:
```
One empty line above `Args:` section

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+
+def change_registry_backend_to_ocs():
+    """
+    Function to deploy registry with OCS backend.
```
Add section:
```
Raises:
    AssertionError: When ...

```

ramkiperiy replied:
ack, will update all the comments related to adding new line, Args, Returns and Raises

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+def get_registry_pod_obj():
+    """
+    Function to get registry pod obj
+    Return:
```
Empty line above Returns: section

Return -> Returns

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+    # i.e. old pod will be terminated and new pod will be up based on new crd
+    # so below loop waits till old pod terminates
+    wait_time = 30
+    while True:
```
while True is danger loop if you don't have specified the number of retries or timeout inside! Please when you use while True you should always do it otherwise can be here fore infinite time when some unexpected behavior happens.

ramkiperiy replied:
ack, WIll add more checks to break the loop.

ramkiperiy replied:
Updated loop with more check to exit.

petr-balogh replied:
Isn't than better option then just use for iteration in range(30):

you can then remove lines:
iteration = 0
iteration += 1

ramkiperiy replied:
ack

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+    Args:
+        role_type (str): Type of the role to be added
+        user (str): User to be added for the role
+    """
```
Add `Raises:` section

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+def validate_pvc_mount_on_registry_pod():
+    """
+    Function to validate pvc mounted on the registry pod
+    """
```
Add `Raises:` section

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+def enable_route_and_create_ca_for_registry_access():
+    """
+    Function to enable route and to create ca and copy to respective location for registry access
+    """
```
Add `Raises:` section

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+
+def enable_route_and_create_ca_for_registry_access():
+    """
+    Function to enable route and to create ca and copy to respective location for registry access
```
Too long line, break it please

ramkiperiy replied:
ack

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+    route = get_default_route_name()
+    with open(f"/tmp/{route}.crt", "wb") as temp:
+        temp.write(base64.b64decode(crt))
+    run_cmd(f"sudo cp /tmp/{route}.crt /etc/pki/ca-trust/source/anchors/{route}.crt")
```
I guess no one wants to have this running on their systems! No sudo commands should be running on the machine where ocs-ci is running and it will fail on the systems when running locally as it will ask for password. In jenkins slaves we don't ask for password so there it will probably work but I think we don't want this to happens there as well.   Can you explain why it's needed and if it can be achieved somehow differently?

Is the route certificate dynamic for each run or is this some specific cert which is same? If second this step should be mentioned as prerequisite and user should do this action by himself. On jenkins we can add it to our playbook with other certs we adding.  (We are using this role: https://gitlab.cee.redhat.com/ocs/ocs4-jenkins/tree/master/ansible/roles/copy-redhat-certs)

ramkiperiy replied:
ack, There is another way to perform this, without using sudo let me explore and update the code.

ramkiperiy replied:
not using sudo any more, resolving this comment.

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+def image_push(image_url, namespace):
+    """
+    Function to push images to destination
+    Args:
```
Empty lines before `Args` and `Returns` sections. Return -> Returns

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+    Function to push images to destination
+    Args:
+        image_url (str): Image url container image repo link
+        namespace {str}: Image to be uploaded namespace
```
(str)

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+    route = get_default_route_name()
+    split_image_url = image_url.split("/")
+    tag_name = split_image_url[-1]
+    tag_cmd = f"podman tag {image_url} {route}/{namespace}/{tag_name}"
```
if you require podman to be installed it needs to be somewhere mentioned as requirement in ocs-ci documentation.  I know that Vaibavi already require this for logging deployment and mentioning it in the docs there, but it's mentioned as dep for logging, so if you need it for some other action it has to be documented as well.

ramkiperiy replied:
Added this command to run outside of master node but yes, as you mentioned podman might not be present. There is a alternate to perform this, default master nodes have podman so there won't be a problem of running podman. Will check and update the code accordingly

ramkiperiy replied:
Update the code, now podman utility is not required in local, all podman cmd will be executed inside master nodes.

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+    crt_dict = ocp_obj.exec_oc_cmd(command=crt_cmd)
+    crt = crt_dict.get('data').get('tls.crt')
+    route = get_default_route_name()
+    with open(f"/tmp/{route}.crt", "wb") as temp:
```
`/tmp/{route}.crt` Use some [tempfile](https://docs.python.org/3/library/tempfile.html) for it directly.

ramkiperiy replied:
ack

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+    route = get_default_route_name()
+    split_image_url = image_url.split("/")
+    tag_name = split_image_url[-1]
+    tag_cmd = f"podman tag {image_url} {route}/{namespace}/{tag_name}"
```
route}/{namespace}/{tag_name} this is used on many places, so define it in variable once and use it instead of construct the same many times.

ramkiperiy replied:
ack

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+def image_list_all():
+    """
+    Function to list the images in the podman registry
+    Return:
```
Empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,209 @@
+    Args:
+        image_url (str): Image url container image repo link
+        namespace {str}: Image to be uploaded namespace
+    Return:
```
Empty line above

petr-balogh commented:
> > @ramkiperiy Is there an issue/ticket describing the requirements and reason for these changes? If so can you link it here please?
> 
> @clacroix12 here is the KNIP https://jira.coreos.com/browse/KNIP-1080

This should be mentioned in the description of PR when opening.

ramkiperiy commented:
> > > @ramkiperiy Is there an issue/ticket describing the requirements and reason for these changes? If so can you link it here please?
> > 
> > 
> > @clacroix12 here is the KNIP https://jira.coreos.com/browse/KNIP-1080
> 
> This should be mentioned in the description of PR when opening.

Updated in singed-off message, do this need to be update in any other place ?

ramkiperiy commented:
@petr-balogh addressed the comments, updated comments sections, removed sudo and don't require podman utility since now all the cmd will be executed inside worker node.
Added more checks for while True loop to exit. would request to review the latest code and if there is any changes required then please comment.

ramkiperiy commented:
Depends on pr #889

vasukulkarni commented:
Please remove 'needs testing' label and add 'verified' label.

petr-balogh commented on ocs_ci/ocs/ocp.py:
```diff
@@ -241,6 +241,24 @@ def apply(self, yaml_file):
+            resource_name (str): Name of the resource
+            params (dict): Changes to be added to the resource
+            type (str): Type of the operation
+        Returns:
```
One empty line above

petr-balogh commented on ocs_ci/ocs/ocp.py:
```diff
@@ -241,6 +241,24 @@ def apply(self, yaml_file):
+    def patch(self, resource_name, params, type='json'):
+        """
+        Applies changes to resources
+        Args:
```
One empty line above

petr-balogh commented on ocs_ci/ocs/ocp.py:
```diff
@@ -241,6 +241,24 @@ def apply(self, yaml_file):
+            type (str): Type of the operation
+        Returns:
+            bool: True in case if changes are applied. False otherwise
+        """
```
One empty line above

petr-balogh commented on ocs_ci/ocs/ocp.py:
```diff
@@ -241,6 +241,24 @@ def apply(self, yaml_file):
+        Returns:
+            bool: True in case if changes are applied. False otherwise
+        """
+        params = "\'" + params + "\'"
```
doesn't look it better with string formatting?
f"\'{params}\'"

petr-balogh commented on ocs_ci/ocs/ocp.py:
```diff
@@ -241,6 +241,24 @@ def apply(self, yaml_file):
         command = f"apply -f {yaml_file}"
         return self.exec_oc_cmd(command)
 
+    def patch(self, resource_name, params, type='json'):
```
Didn't comment this method as you pointed that it's taken from other PR but as it looks this one will be merged soon I have some comments also for this method.

ramkiperiy replied:
ack

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+        pod_data = pod.get_pods_having_label(
+            label='docker-registry=default', namespace=constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE
+        )
+        pod_obj = [pod.Pod(**data) for data in pod_data]
```
pod object looks like the list to me here and you are saying in the doc that you are returning 
pod_obj (obj): Registry pod obj
But on the line 75 seems you are returning this list of pod objects. Isn't that suppose to return pod_obj[0] then? Or did I miss something?

ramkiperiy replied:
While changing the registry backend for some seconds or minutes there will be 2 pods running i.e. Existing one which will be in terminating and new pod which will be Up. Expectation is only one pod obj but during the backend change process there will be 2 pods.

ramkiperiy replied:
ack, updated the Returns comment for line 75

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+    )
+    with open(filename) as f:
+        password = f.read()
+    helpers.refresh_oc_login_connection()
```
Didn't get why we need to do this?  Once we execute the run-ci we have exported kubeconfig so why this is needed at all?

ramkiperiy replied:
Observed problems to get_user_token if login is not refreshed. due to this podman login will fail.

petr-balogh replied:
So is this needed just to obtain the token?

Cannot we use this approach:
https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/

```
APISERVER=$(oc config view --minify -o jsonpath='{.clusters[0].cluster.server}')
SECRET_NAME=$(oc get serviceaccount default -o jsonpath='{.secrets[0].name}')
TOKEN=$(oc get secret $SECRET_NAME -o jsonpath='{.data.token}' | base64 --decode)
```

I tried above and then:
`curl $APISERVER/api --header "Authorization: Bearer $TOKEN" --insecure`
worked for me.

ramkiperiy replied:
Will test the code with the changes and update sooner.

petr-balogh replied:
In first place I am not sure if this token is that one we really need :) just when I looked to google how to obtain the token that was what I found but not sure if this is the same token as we can see from whoami command.  If you can confirm that this token is what we need it can be part of that function but I hope that maybe @fbalak can shed some light here as git blame wrote that he wrote that function. So maybe @fbalak can lear us something more about this.

fbalak replied:
@petr-balogh This is different token. I tried token from https://github.com/red-hat-storage/ocs-ci/pull/868/files/2387da9a0b97fd8fcf81332556059a475086bfa6#r334872621 generated from secret as you wrote but it didn't work when I tried to use it with prometheus API.

ramkiperiy replied:
With the $TOKEN podman login failed.
```
podman login default-route-openshift-image-registry.apps.rperiyas-cls.qe.rh-ocs.com -u kubeadmin -p ZXlKaGJHY2lPaUpTVXpJMU5pSXNJbXRwWkNJNklpSjkuZXlKcGMzTWlPaUpyZFdKbGNtNWxkR1Z6TDNObGNuWnBZMlZoWTJOdmRXNTBJaXdpYTNWaVpYSnVaWFJsY3k1cGJ5OXpaWEoyYVdObFlXTmpiM1Z1ZEM5dVlXMWxjM0JoWTJVaU9pSmtaV1poZFd4MElpd2lhM1ZpWlhKdVpYUmxjeTVwYnk5elpYSjJhV05sWVdOamIzVnVkQzl6WldOeVpYUXVibUZ0WlNJNkltUmxabUYxYkhRdGRHOXJaVzR0Y3pkaU5Hc2lMQ0pyZFdKbGNtNWxkR1Z6TG1sdkwzTmxjblpwWTJWaFkyTnZkVzUwTDNObGNuWnBZMlV0WVdOamIzVnVkQzV1WVcxbElqb2laR1ZtWVhWc2RDSXNJbXQxWW1WeWJtVjBaWE11YVc4dmMyVnlkbWxqWldGalkyOTFiblF2YzJWeWRtbGpaUzFoWTJOdmRXNTBMblZwWkNJNklqRTBZekl3WlRGaUxXVm1ORFV0TVRGbE9TMDRPVFprTFRBeU16SXpOR0k1WmpFeU9DSXNJbk4xWWlJNkluTjVjM1JsYlRwelpYSjJhV05sWVdOamIzVnVkRHBrWldaaGRXeDBPbVJsWm1GMWJIUWlmUS5CNGF3RGJ0VC0xeHRzVnlqSGtTRFhGUzF5R0kxTE1vdmk2bEdJN284d1FlZXRIaFJUT2FVWjFDaHpFbFA4X0tNTVlSRlE2RTh0UVZLdDdFTVVXSFV6bHZVaXNzSmprMjBaQ0VyUS1KejNpZTd4ZWpuVDVQa0g1S2k1WlZpay0xUkVJOXpVMGxFWmo5alNHeU9mU21UUWhLaEt4Qm9Ja1FvbG41bEJ2VkpydnFNcXdGOUI2MFBJSzlZSTR3VDBsQ3dLVU5aZ0VBNzhIbEsxZnB0dTZfeFhkUnhydEYteFdwZXQ1Tk81UDkxZUx2d1VIbnBMNW5TTFFkelhDYnNMNEdwRlN3Sll6M3ZPSlhUcWcyWGJOUUtsZWhMSTNHQUJGVE9fZktFMnBkZUF3QmhxLVRjNjNTTUhyUUdHdGFjeXVUUkwwTjVESm02LXVMVVhUalB4NXVhelE=

Error: error logging into "default-route-openshift-image-registry.apps.rperiyas-cls.qe.rh-ocs.com": invalid username/password
'''

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+    route = get_default_route_name()
+    cmd_list = [
+        'export KUBECONFIG=/home/core/auth/kubeconfig',
+        f"oc login -u {user} -p {password}",
```
If we have kubeconfig exported why the login is needed?

ramkiperiy replied:
This set of commands will be executed inside the NODE. In debug mode kubeconfig export, oc login is required for each podman command execution, else podman commands will fail.

petr-balogh commented on tests/helpers.py:
```diff
@@ -1352,3 +1367,53 @@ def get_memory_leak_median_value():
+    )
+    str_split = filename.split('/')
+    str_split = str_split[:-1]
+    file_path = '/'.join(str_split)
```
Are you trying to achieve here something which os.path.dirname function already provides to you? :)

ramkiperiy replied:
default filename provides file path '/home/rperiyas/cluster/auth/kubeconfig' , this can't be used in rsync so to remove the kubeconfig. splitting the string using '/' and removing last list and joining again using '/', so that rsync can sync the directory.

petr-balogh replied:
I understand but look:  
``` python
os.path.dirname("/home/rperiyas/cluster/auth/kubeconfig")
# OUTPUT VALUE:
'/home/rperiyas/cluster/auth'
```
This is what you want no?

ramkiperiy replied:
yes

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+
+    Raises:
+        AssertionError: When PVC mount not present in the registry pod
+    """
```
One empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+
+    Raises:
+        UnexpectedBehaviour: When image-registry pod is not present.
+    """
```
One empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+
+    Returns:
+        pvc_name (str): Returns name of the OCS pvc backed for registry
+    """
```
One empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+    """
+    Function to validate all image-registry pod status
+    """
+    pass
```
Raise NotImplementedError when it's not implemented yet ore just remove this function from this PR

ramkiperiy replied:
ack

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+    """
+    Function to validate image_push data from ceph
+    """
+    pass
```
Raise NotImplementedError when it's not implemented yet ore just remove this function from this PR

ramkiperiy replied:
ack

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+
+    Raises:
+        AssertionError: When failure in adding new role to user
+    """
```
One empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+        AssertionError: When failure in adding new role to user
+    """
+    ocp_obj = ocp.OCP()
+    role_cmd = f"policy add-role-to-user {role_type} {user} -n {constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE}"
```
too long line

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+    """
+    ocp_obj = ocp.OCP()
+    role_cmd = f"policy add-role-to-user {role_type} {user} -n {constants.OPENSHIFT_IMAGE_REGISTRY_NAMESPACE}"
+    assert ocp_obj.exec_oc_cmd(command=role_cmd), f"Adding role failed"
```
I see you are trying format string but no variable formatted. f"Adding role failed" . So remove f if no need to format here

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+
+    Raises:
+        AssertionError: When failure in enabling registry default route
+    """
```
One empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+
+    Args:
+        image_url (str): Image url container image repo link
+    """
```
One empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+
+    Returns:
+        registry_path (str): Uploaded image path
+    """
```
One empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+
+    Returns:
+        image_list_output (str): Images present in cluster
+    """
```
One empty line above

petr-balogh commented on ocs_ci/ocs/registry.py:
```diff
@@ -0,0 +1,282 @@
+
+    Args:
+        registry_path (str): Image registry path
+    """
```
one empty line above

petr-balogh commented on tests/helpers.py:
```diff
@@ -768,6 +769,20 @@ def get_worker_nodes():
+    Fetches all master nodes.
+
+    Returns:
+        list: List of names of worker nodes
```
master nodes

petr-balogh commented on tests/helpers.py:
```diff
@@ -1352,3 +1367,53 @@ def get_memory_leak_median_value():
+    ocp_obj = ocp.OCP()
+    check_auth = 'auth'
+    check_conf = 'kubeconfig'
+    if check_auth not in ocp_obj.exec_oc_debug_cmd(node=master_list[0], cmd_list=['ls /home/core/']):
```
/home/core/ is used three times, define it in some variable and use it from there.

ramkiperiy replied:
ack

ramkiperiy commented:
Local run logs 
```
15:04:03 - MainThread - ocs_ci.ocs.resources.ocs - INFO - Adding PersistentVolumeClaim with name registry-cephfs-rwx-pvc
15:04:03 - MainThread - ocs_ci.utility.templating - INFO - apiVersion: v1
                                                           kind: PersistentVolumeClaim
                                                           metadata:
                                                             name: registry-cephfs-rwx-pvc
                                                             namespace: openshift-image-registry
                                                           spec:
                                                             accessModes:
                                                             - ReadWriteMany
                                                             resources:
                                                               requests:
                                                                 storage: 100Gi
                                                             storageClassName: rook-ceph-cephfs
15:04:03 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig create -f /tmp/PersistentVolumeClaim2a6ajq_k -o yaml
15:04:04 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig get PersistentVolumeClaim registry-cephfs-rwx-pvc -n openshift-image-registry -o yaml
15:04:05 - MainThread - ocs_ci.ocs.ocp - INFO - Waiting for a resource(s) of kind PersistentVolumeClaim identified by name 'registry-cephfs-rwx-pvc' and selector None to reach desired condition Bound
15:04:05 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig get PersistentVolumeClaim registry-cephfs-rwx-pvc -n openshift-image-registry -o yaml
15:04:06 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig get PersistentVolumeClaim registry-cephfs-rwx-pvc -n openshift-image-registry
15:04:08 - MainThread - tests.helpers - INFO - PersistentVolumeClaim registry-cephfs-rwx-pvc reached state Bound
15:04:08 - MainThread - ocs_ci.ocs.ocp - INFO - Command: patch Config cluster -n openshift-image-registry -p '[{"op": "add", "path": "/spec/storage", "value": {"pvc": {"claim": "registry-cephfs-rwx-pvc"}}}]' --type json
15:04:08 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig patch Config cluster -n openshift-image-registry -p '[{"op": "add", "path": "/spec/storage", "value": {"pvc": {"claim": "registry-cephfs-rwx-pvc"}}}]' --type json
15:04:09 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig get Pod  -n openshift-image-registry --selector=docker-registry=default -o yaml
15:04:11 - MainThread - ocs_ci.ocs.registry - INFO - Waiting for 30 sec's for registry pod to be up iteration 1
15:04:41 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig get Pod  -n openshift-image-registry --selector=docker-registry=default -o yaml
15:04:42 - MainThread - ocs_ci.ocs.ocp - INFO - Waiting for a resource(s) of kind Pod identified by name 'image-registry-7f4ccbfb69-86d8x' and selector None to reach desired condition Running
15:04:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig get Pod image-registry-7f4ccbfb69-86d8x -n openshift-image-registry -o yaml
15:04:43 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig get Pod image-registry-7f4ccbfb69-86d8x -n openshift-image-registry
15:04:44 - MainThread - tests.helpers - INFO - Pod image-registry-7f4ccbfb69-86d8x reached state Running
15:04:44 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig get Pod  -n openshift-image-registry --selector=docker-registry=default -o yaml
15:04:45 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-image-registry --kubeconfig /home/rperiyas/15thOct/auth/kubeconfig rsh image-registry-7f4ccbfb69-86d8x mount
15:04:49 - MainThread - ocs_ci.ocs.registry - INFO - Verified pvc is mounted on image-registry pod
15:04:49 - MainThread - ocs_ci.deployment.deployment - INFO - Done creating rook resources, waiting for HEALTH_OK
```
Full log available in this pastebin http://pastebin.test.redhat.com/805873

vasukulkarni commented:
@petr-balogh I am going to dismiss your review, since all comments are addressed and test is working fine.

vasukulkarni commented:
@ramkiperiy lets run one final job with jenkins, thanks
